util/quickstart/setchplenv.fish

```diff
@@ -1,36 +1,40 @@
 # fish shell script to set the Chapel environment variables
 
-# shallow test to see if we are in the correct directory
+# Directory of setchplenv script, will not work if script is a symlink
+set DIR (cd (dirname (status -f)); pwd)
+
+# Shallow test to see if we are in the correct directory
 # Just probe to see if we have a few essential subdirectories --
 # indicating that we are probably in a Chapel root directory.
-if [ ! -d "util" -o ! -d "compiler" -o ! -d "runtime" -o ! -d "modules" ]
-	echo "Error: You must use '. util/setchplenv' from within the chapel root directory."
-	exit
+set chpl_home (fish -c "cd $DIR/../../; pwd")
+if [ ! -d "$chpl_home/util" -o ! -d "$chpl_home/compiler" -o ! -d "$chpl_home/runtime" -o ! -d "$chpl_home/modules" ]
+    echo "Error: \$CHPL_HOME is not where it is expected"
+    exit
 end
 
-eval set MYPATH (./util/config/fixpath "$PATH" " ")
-eval set MYMANPATH (./util/config/fixpath "$MANPATH" " ")
+echo -n "Setting CHPL_HOME "
+set -x CHPL_HOME $chpl_home
+echo "to $CHPL_HOME"
+
+set MYPATH (eval "$CHPL_HOME/util/config/fixpath.py \"PATH\" \"fish\"")
+set MYMANPATH (eval "$CHPL_HOME/util/config/fixpath.py \"MANPATH\" \"fish\"")
 
 if [ (count $MYPATH) = 0 ]
-  echo "Error running ./util/config/fixpath"
+  echo "Error running \$CHPL_HOME/util/config/fixpath"
   exit
 end
 
-echo -n "Setting CHPL_HOME "
-set -x CHPL_HOME $PWD
-echo "to $CHPL_HOME"
-
 echo -n "Setting CHPL_HOST_PLATFORM "
-set -x CHPL_HOST_PLATFORM (eval $CHPL_HOME/util/chplenv/chpl_platform.py)
+set -x CHPL_HOST_PLATFORM (eval "$CHPL_HOME/util/chplenv/chpl_platform.py")
 echo "to $CHPL_HOST_PLATFORM"
 
 echo -n "Updating PATH "
-set -x PATH "$CHPL_HOME/bin/$CHPL_HOST_PLATFORM" "$CHPL_HOME/util" $MYPATH
+eval set -x PATH $CHPL_HOME/bin/$CHPL_HOST_PLATFORM $CHPL_HOME/util $MYPATH
```

Comment:
I'd prefer the `eval`s be moved up to `set MYPATH` and `set MYMANPATH` lines, it is clearer why they are required when placed there.